packages/core/src/http-client.ts

```diff
@@ -91,3 +96,24 @@ function mergeWithDefaultRequestConfig(
     params: mergeIgnoreCase(defaultConfig.params, requestConfig?.params)
   };
 }
+
+/**
+ * Get target url with endpoint path appended.
+ * Append path `v2` if the url contains empty pathname `/`.
+ * @param url - The url, e.g., `http://example.com` or `http://example.com:8000/abc`.
+ * @param endpointPath - The path to the endpoint, e.g., `/some/endpoint`.
+ * @returns Target url combining the url and endpoint path.
+ * @internal
+ */
```

Comment:
[q] Is Java also adapting this?

Reply:
Yes, this is a JS translation from Java 😄